ptf

```diff
@@ -220,6 +221,21 @@ be subtracted from the result by prefixing them with the '^' character.  """
                 )
             setattr(namespace, self.dest, values)
 
+            
+    class ActionGroupMode(argparse.Action):
+        """
+        Check the group-mode parameters.
+        """
+        def __call__(self, parser, namespace, values, option_string=None):
+            parameters = ['class', 'method']
+            assert type(values) is str
+            if not values in parameters:
+                parser.error(
+                    "invalid value for --group-mode: value %s does not support in %s" % (values, parameters)
+                )
+            setattr(namespace, self.dest, values)
```

Comment:
The `add_argument` method supports the `choices` parameter which does the same thing, there is no need to write custom validation logic for this.

Reply:
thanks for the suggestion